core/templates/pages/exploration-editor-page/services/exploration-param-changes.service.ts

```diff
@@ -17,13 +17,28 @@
  * changes to parameters.
  */
 
-require(
-  'pages/exploration-editor-page/services/exploration-property.service.ts');
+import { ExplorationPropertyService } from 'pages/exploration-editor-page/services/exploration-property.service';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { AlertsService } from 'services/alerts.service';
+import { ChangeListService } from './change-list.service';
+import { LoggerService } from 'services/contextual/logger.service';
```

Comment:
why are these needed?

Reply:
i guess the ExplorationPropertyService needs them as the service below extends it. And since ExplorationPropertyService is not actually decorated with an @Injectable decorator, we have to pass those singletons ourselves.